src/Engine/ProtoCore/DSASM/Executive.cs

```diff
@@ -108,6 +108,96 @@ public Executive(Core core, bool isFep = false)
+                StackFrameType callerFrameType = stackFrame.CallerStackFrameType;
+                StackFrameType frameType = stackFrame.StackFrameType;
+                Validity.Assert(frameType == StackFrameType.kTypeLanguage);
+
+                int depth = stackFrame.Depth;
+                int framePointer = stackFrame.FramePointer;
+                List<StackValue> registers = stackFrame.GetRegisters();
+
+                rmem.PushStackFrame(svThisPtr, ci, fi, returnAddr, blockDecl, blockCaller, callerFrameType, frameType, depth + 1, framePointer, registers, locals, 0);
+            }
+
+            StackValue sv = core.ExecutionInstance.Execute(exeblock, entry, context, sink);
+            return sv;
+        }
+
+        /// <summary>
+        /// Bounce instantiates a new Executive 
+        /// Execution jumps to the new executive
+        /// This iverload handles debugger properties when bouncing
+        /// </summary>
+        /// <param name="exeblock"></param>
+        /// <param name="entry"></param>
+        /// <param name="context"></param>
+        /// <param name="breakpoints"></param>
+        /// <param name="stackFrame"></param>
+        /// <param name="locals"></param>
+        /// <param name="exec"></param>
+        /// <param name="fepRun"></param>
+        /// <returns></returns>
+        public StackValue Bounce(
+            int exeblock, 
+            int entry, 
+            ProtoCore.Runtime.Context context, 
+            List<Instruction> breakpoints, 
+            StackFrame stackFrame, 
+            int locals = 0,
+            DSASM.Executive exec = null,
+            ProtoCore.DebugServices.EventSink sink = null,
+            bool fepRun = false)
+        {
+            if (stackFrame != null)
+            {
+                StackValue svThisPtr = stackFrame.ThisPtr;
+                int ci = stackFrame.ClassScope;
+                int fi = stackFrame.FunctionScope;
+                int returnAddr = stackFrame.ReturnPC;
+                int blockDecl = stackFrame.FunctionBlock;
+                int blockCaller = stackFrame.FunctionCallerBlock;
+                StackFrameType callerFrameType = stackFrame.CallerStackFrameType;
+                StackFrameType frameType = stackFrame.StackFrameType;
+                Validity.Assert(frameType == StackFrameType.kTypeLanguage);
+                int depth = stackFrame.Depth;
+                int framePointer = stackFrame.FramePointer;
+                List<StackValue> registers = stackFrame.GetRegisters();
+
+                core.DebugProps.SetUpBounce(exec, blockCaller, returnAddr);
+
+                rmem.PushStackFrame(svThisPtr, ci, fi, returnAddr, blockDecl, blockCaller, callerFrameType, frameType, depth + 1, framePointer, registers, locals, 0);
+            }
+            StackValue sv = core.ExecutionInstance.Execute(exeblock, entry, context, breakpoints, sink, fepRun);
+            return sv;
```

Comment:
Done, ive removed the overloads for debug and have only one coherent set of functions to handle both run and debug (Bounce, Run, Execute )